File: Nivelamento/C-Cpp-Java/Repeticao.cpp
#include <iostream>

using namespace std;

int main(void) {
	float base, altura, area;
	//
	cout << "Digite 0 para parar." << endl;
	//
	do{
		cout << "Digite a base do retangulo: ";
		cin >> base;
		//
		// nao eh a maneira mais eficiente
		if (base > 0){
			cout <<	"Digite a altura do retangulo: ";
			cin >> altura;
			//
			area = base * altura;
			//
			cout << "A area do retangulo = " << area << endl;
		}
	}while (base > 0);
	//
	return 0;
}

File: Nivelamento/C-Cpp-Java/Repeticao.java
class Repeticao{
	public static void main (String[] agrs){
		int area,
		    i =1;
		//
		// base inicia em 1
		// repetira enquando <=5
		// e incrementara de 1 em 1
		for (int base = 1; base <= 5; base++){
			// inicia altura como 2
			// repetira enquando altura <=10
			// altura incrementara de 2 em 2
			for (int altura = 2; altura <= 10; altura += 2, i++){
				// esta utilizando a propria variavel i para incrementar e atualizar
				//  os retangulos
				//
				area = base * altura;
				//
				System.out.println(i+" retangulo");
				System.out.println("============");
				System.out.println("Base = "+base+" e altura = "+altura);
				System.out.println("Area = "+area+"\n");
			}
		}
	}
}
